docs/en/integrations/language-clients/java/client-v2.md

```diff
@@ -0,0 +1,539 @@
+---
+sidebar_label: Client V2
+sidebar_position: 2
+keywords: [clickhouse, java, jdbc, client, integrate, r2dbc]
+description: Options for connecting to ClickHouse from Java
+slug: /en/integrations/java/client-v2
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+
+# Java Client (V2)
+
+Implementation of a new API. It uses Apache Http Client to communicate with ClickHouse server. We have selected this http client because it has many built-in features and 
+has proven itself in old client implementation. We are planning to support other http client libraries. 
+
+*Note*: Client-V2 is currently in the phase of active development and we are still working on it. 
+
+## Setup
+
+- Maven Central (project web page): https://mvnrepository.com/artifact/com.clickhouse/client-v2
+- Nightly builds (repository link): https://s01.oss.sonatype.org/content/repositories/snapshots/com/clickhouse/
+
+<Tabs groupId="client-v2-setup">
+<TabItem value="maven" label="Maven" >
+
+```xml 
+<dependency>
+    <groupId>com.clickhouse</groupId>
+    <artifactId>client-v2</artifactId>
+    <version>0.6.5</version>
+</dependency>
+```
+
+</TabItem>
+<TabItem value="gradle-kt" label="Gradle (Kotlin)">
+
+```kotlin
+// https://mvnrepository.com/artifact/com.clickhouse/client-v2
+implementation("com.clickhouse:client-v2:0.6.5")
+```
+</TabItem>
+<TabItem value="gradle" label="Gradle">
+
+```groovy
+// https://mvnrepository.com/artifact/com.clickhouse/client-v2
+implementation 'com.clickhouse:client-v2:0.6.5'
+```
+
+</TabItem>
+</Tabs>
+
+## Initialization
+
+Client object is initialized by `com.clickhouse.client.api.Client.Builder#build()`. Each client has own context and no objects are shared between them.
+Builder has configuration method for convinient setup. 
+
+Example: 
+```java showLineNumbers
+ Client client = new Client.Builder()
+                .addEndpoint("https://clickhouse-cloud-instance:8443/")
+                .setUsername(user)
+                .setPassword(password)
+                .build;
+```
+
+`Client` is `AutoCloseable` and should be closed when not needed anymore. 
+
+## Configuration 
+
+All settings are defined by instance methods (a.k.a configuration methods) that make scope and context of each value clear. 
+Major configuration parameters are defined in one scope (client or operation) and do not override each other. Handling 
+configuration overriding across is a very hard task so we doing our best to keep it simple. 
```

Comment:
`Handling configuration overriding across is a very hard task so we doing our best to keep it simple.` 
I would not state it here